airbyte_cdk/sources/declarative/resolvers/http_components_resolver.py

```diff
@@ -88,19 +88,25 @@ def resolve_components(
         """
         kwargs = {"stream_template_config": stream_template_config}
 
-        for components_values in self.retriever.read_records({}):
-            updated_config = deepcopy(stream_template_config)
-            kwargs["components_values"] = components_values  # type: ignore[assignment] # component_values will always be of type Mapping[str, Any]
+        stream_slices = self.retriever.stream_slices() if self.retriever.stream_slicer else [{}]
```

Comment:
Do you know of any cases where `stream_slicer` is not present in `retriever`?

https://github.com/airbytehq/airbyte-python-cdk/blob/adef1e8fb38fb792825c6a4c08d8e81b0fab3851/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py#L82

I also think that using `[None]` is safer.

Reply:
I don't know a case when stream_slicer not present in the retriever, also  default stream slicer that you are referring to uses single partition router [here](https://github.com/airbytehq/airbyte-python-cdk/blob/adef1e8fb38fb792825c6a4c08d8e81b0fab3851/airbyte_cdk/sources/declarative/partition_routers/single_partition_router.py#L51). So I think we can go without the condition at all.
 I'll update the code.